submissions/alstep07/task-js-post-oop/index.js

```diff
@@ -0,0 +1,116 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: _put repo URL here_
+   Web app: _put project's github pages URL here_
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+
+class Inhabitant {
+	constructor(species, name, gender, saying, friends = []) {
+		this.species = species;
+		this.name = name;
+		this.gender = gender;
+		this.saying = saying;
+		this.friends = friends;
+	}
+
+	toString() {
+		let greating = [
+			`${this.saying}`,
+			`I am a ${this.species}`,
+			`My gender is ${this.gender}`,
+			`My name is ${this.name}`,
+		];
+
+		if (this.friends.length > 1) {
+			greating.push(`My friends are ${this.friends.join(", ")}`);
+		} else if (this.friends.length === 1) {
+			greating.push(`My friend is ${this.friends}`);
+		}
+		return greating.join(". ");
+	}
+}
+
+class Animal extends Inhabitant {
+	constructor(species, name, gender, saying, friends, legs) {
+		super(species, name, gender, saying, friends);
+		this.legs = legs;
+	}
+	toString() {
+		return [super.toString(), `I have ${this.legs} legs`].join(". ");
+	}
+}
+
+class Human extends Animal {
+	constructor(name, gender, saying, friends) {
+		super("human", name, gender, saying, friends);
+		this.legs = 2;
```

Comment:
Perhaps you wanted to send legs count via `super()` just like this is done for dogs and cats?

Reply:
@OleksiyRudenko You are right, I just hurried and forgot to fix the similar code snippet.